packages/block-library/src/image/edit.js

```diff
@@ -177,10 +198,29 @@ class ImageEdit extends Component {
 			isEditing: false,
 		} );
 
+		let src = media.url;
+		let img = {};
+		let fileWidth;
+		let fileHeight;
+
+		if ( media.sizes ) {
+			// The "full" size is already included in `sizes`.
+			img = media.sizes.large || media.sizes.full;
+			src = img.url;
+			fileWidth = get( img, [ 'actual_size', 'width' ] ) || img.width;
+			fileHeight = get( img, [ 'actual_size', 'height' ] ) || img.height;
+		}
+
+		const attr = pickRelevantMediaFiles( media );
```

Comment:
why are we pulling this out? it's still confusing me and makes me think there's something else going on even though to the best of my knowledge there's nothing happening. we can still set the `url` attributes in the `setAttributes()` call just by adding it after `...pickRelevantMediaFiles( media )` and then it won't be splitting the computation or providing a point where we could introduce a data consistency bug

Reply:
Hm, this is actually not needed any more as now `pickRelevantMediaFiles()` properly gets the "large" size of the image if it exists. That was the same "fix".